yourgov/components/FormMobileFoodVendorPermit/utils.ts

```diff
@@ -0,0 +1,36 @@
+import * as yup from 'yup';
+import { parseISO, isValid } from 'date-fns';
+
+// `yup.date()` can sometimes give false positives with certain string values
+// Fixes https://github.com/jquense/yup/issues/764
+// Tests for this can be found in `packages/react/src/date-picker/test-utils.ts`
+export const yupDateField = yup
+	.date()
+	.transform((current, orig) => (typeof orig === 'string' ? orig : current))
+	.typeError('Enter a valid date');
+
+// Session storage converts converts dates to ISO strings, so we need to convert these values back to dates
+export function parseDateField(value: Date | string | undefined) {
+	if (typeof value === 'string') {
+		const parsedValue = parseISO(value);
+		return isValid(parsedValue) ? parsedValue : undefined;
+	}
+	return value;
+}
+
+export const yupPhoneField = yup
+	.string()
+	.transform((current) => current.replace(/ /g, ''))
+	.matches(/^\d+$/, {
+		excludeEmptyString: true,
+		message: 'Number must not include letters or symbols',
+	})
+	.matches(/^0\d+$/, {
+		excludeEmptyString: true,
+		message:
+			"Mobile numbers must begin with '04', landline numbers must begin with an area code",
+	})
+	.matches(/^\d+.{9}$/, {
```

Comment:
Hmmmm this let me add an 18 digit number

Reply:
Whoa, good catch! Since we do want to allow white spaces, this regex is a little bit tricky. But I've fixed this bug, and I've also added a bunch of tests too https://github.com/agriculturegovau/agds-next/blob/b6e7e351756a20ca61cabfcebbf3ac442403baaa/yourgov/components/FormMobileFoodVendorPermit/utils.test.ts